stock_picking_force_assign/models/stock_picking.py

```diff
@@ -70,3 +76,23 @@ def _force_assign_find_moves(self):
                     )
                 )
         return result
+
+    def _unchain_waiting(self, link_template):
+        """Set moves to make_to_stock and unlink move_orig_ids from moves.
+        Moves must be unlinked because _action_assign will otherwise only
+        consider availability in linked moves, instead of considering the
+        availability at the location.
+        Moves that have procure_method set to make_to_order will also
+        be skipped by _action_assign, so we set it to make_to_stock.
+        """
+        res = False
+        to_unchain = self.move_lines.mapped('move_orig_ids')
+        if to_unchain:
+            self.message_post(body=_(
+                'Decoupled operation(s) %s in order to assign this one'
+            ) % ', '.join(to_unchain.mapped(
+                lambda move: link_template % (move._name, move.id, move.name)
+            )))
+        if to_unchain or any(line.procure_method != 'make_to_stock' for line in self.move_lines):
+            res = self.mapped('move_lines').write({'procure_method': 'make_to_stock', 'move_orig_ids': [(5, 0, 0)]})
```

Comment:
also here better collect the lines where you need the write, and only do it on them